src/snmp.c

```diff
@@ -1397,6 +1397,38 @@ static int csnmp_read_table(host_definition_t *host, data_definition_t *data) {
       break;
     }
 
+    if (res->errstat != SNMP_ERR_NOERROR) {
+      if (res->errindex != 0 && res->errindex < oid_list_len) {
```

Comment:
This if-block is unnecessarily long, please invert this logic keep things short:

```c
if ((res->errindex == 0) || (res->errindex >= oid_list_len)) {
  ERROR("snmp plugin: host %s; data %s: response error: %s (%li) ",
        host->name, data->name, snmp_errstring(res->errstat), res->errstat);
  status = -1;
  break;
}
/* previous if-block. */
```